homeassistant/components/template/template_entity.py

```diff
@@ -221,3 +222,38 @@ async def async_added_to_hass(self):
             )
 
         await super().async_added_to_hass()
+
+
+class TemplateEntityWithAvailabilityAndImages(TemplateEntityWithAvailability):
+    """Entity that uses templates to calculate attributes with an availability, icon, and images template."""
+
+    def __init__(self, availability_template, icon_template, entity_picture_template):
```

Comment:
Wouldn't it make sense to just have 1 generic TemplateEntity that uses kwargs for availability, icon, entity_picture and just handles all generic properties ? 

```suggestion
    def __init__(self, *, availability_template=None, icon_template=None, entity_picture_template=None):
```

Reply:
I realized how organic these were after I did most of the conversions.  I'll collapse them all into a generic `TemplateEntity` in a followup once all the conflicts are fixed

Reply:
done here https://github.com/home-assistant/core/pull/39083